h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -713,13 +730,16 @@ public final double deviance(double yr, double ym){
           if( yr == 0 ) return -2;
           return -2 * (Math.log(yr / ym) - (yr - ym) / ym);
         case tweedie:
-          double theta = _var_power == 1
-            ?Math.log(y1/ym)
-            :(Math.pow(y1,1.-_var_power) - Math.pow(ym,1 - _var_power))/(1-_var_power);
-          double kappa = _var_power == 2
-            ?Math.log(y1/ym)
-            :(Math.pow(yr,2-_var_power) - Math.pow(ym,2-_var_power))/(2 - _var_power);
-          return 2 * (yr * theta - kappa);
+          double val;
+          if (_var_power==1) {
+            val = yr*Math.log(y1/ym)-(yr-ym);
+          } else if (_var_power==2) {
+            val = yr*(1/ym-1/y1)-Math.log(y1/ym);
+          } else {
+            val = yr*_oneOoneMinusVarPower*(Math.pow(yr,_oneMinusVarPower)-Math.pow(ym, _oneMinusVarPower))-
+                    (Math.pow(yr,_twoMinusVarPower)-Math.pow(ym, _twoMinusVarPower))*_oneOtwoMinusVarPower;
+          }
+          return 2 * val;
         default:
```

Comment:
I am calculating the deviance according to this formula:

<img width="424" alt="image" src="https://user-images.githubusercontent.com/7231712/58065964-5c66b580-7b3c-11e9-9c6c-87986524ff8b.png">